nodemapper/src/gui/Builder/components/Flow.tsx

```diff
@@ -140,81 +141,118 @@ const ModuleNode = (props: NodeProps<ModuleData>) => {
-          className={styles.BodyPanel}
-          style={{
-            height: `${input_namespaces.length * 18 - 4}px`,
-          }}
-        >
-          {input_namespaces.map((name) => (
-            <div key={"div-" + name}>
+    <>
+      <div
+        className={styles.Node}
+        style={{
+          backgroundColor: props.data.color,
+        }}
+      >
+        {selected ? (
+          <NodeResizeControl
+            style={nodeResizeControlStyle}
+            minWidth={120}
+            minHeight={50}
+            variant={ResizeControlVariant.Line}
+            position="top-right"
+            shouldResize={(e, params) => params.direction[1] === 0}
+          >
+            <span
+              style={{
+                float: "right",
+                color: "#bebebe",
+              }}
+            >
+              <FontAwesomeIcon icon={faLeftRight} />
+            </span>
+          </NodeResizeControl>
+        ) : null}
+        <div className={styles.HeaderPanel}>
+          <div className={styles.HeaderText}>{props.data.config.name}</div>
+        </div>
+        {!named_inputs ? (
+          <>
+            {input_namespaces.length == 1 ? (
               <Handle
                 className={styles.HandleInput}
-                id={name}
-                key={name}
+                id={input_namespaces[0]}
+                key={input_namespaces[0]}
                 type="target"
                 position={Position.Left}
-                style={{ top: `${input_namespaces.indexOf(name) * 18 + 38}px` }}
-                onClick={(event) => {
-                  event.stopPropagation();
-                  console.log("Input handle clicked: ", event.target);
-                }}
-              >
-                <div
-                  className={styles.InputPortLabel}
-                  style={{
-                    pointerEvents: "none", // pass-through click events
-                  }}
-                >
-                  {name}
-                </div>
-              </Handle>
-            </div>
-          ))}
-          <div>
+                style={{ top: "50%" }}
+              />
+            ) : null}
             <Handle
               className={styles.HandleOutput}
               id="out"
               type="source"
               position={Position.Right}
-              style={{
-                top: `${((input_namespaces.length - 1) / 2) * 18 + 38}px`,
-              }}
+              style={{ top: "50%" }}
             />
+          </>
+        ) : (
+          <div
+            className={styles.BodyPanel}
+            style={{
+              height: `${input_namespaces.length * 18 - 4}px`,
+            }}
+          >
+            {input_namespaces.map((name) => {
+              // Format port name
+              const port_name_split = name.split("$");
+              let port_name = node_config[port_name_split[0]]?.name ?? null;
+              if (port_name === undefined || port_name === null) {
```

Comment:
Is there a reason to use this if instead of just this?
```suggestion
              if (!port_name) {
```